app/bundles/EmailBundle/Form/Type/ExampleSendType.php

```diff
@@ -39,5 +63,17 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'save_icon'  => 'fa fa-send',
             ]
         );
+
+        if (!empty($options['action'])) {
+            $builder->setAction($options['action']);
+        }
+    }
+
+    /**
+     * @param OptionsResolverInterface $resolver
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $resolver->setRequired(['leads']);
```

Comment:
This may be considered as a BC break because whoever is using this form will now have to add new option.

I think that it would be better to load the contacts in this form [instead of the controller](https://github.com/mautic/mautic/pull/6090/files#diff-dd63f74bd8ab68313aab62a5bef99db1R1441) and set optional `$filter` to search for the right contacts.

Reply:
It's better design to load contacts directly in the form. If this form would be used from another controller than this controller would have to prepare the contact list too.

Reply:
Okay, make sense that ;)

Reply:
Update, hope ready to test now ;)